core/ir/opnd_shared.c

```diff
@@ -2621,8 +2621,13 @@ reg_get_size(reg_id_t reg)
         return OPSZ_4;
     if (reg >= DR_REG_MDCCSR_EL0 && reg <= DR_REG_SPSR_FIQ)
         return OPSZ_8;
-    if (reg >= DR_REG_Z0 && reg <= DR_REG_Z31)
+    if (reg >= DR_REG_Z0 && reg <= DR_REG_Z31) {
+#        if !defined(DR_HOST_NOT_TARGET) && !defined(STANDALONE_DECODER)
```

Comment:
For standalone decoder we need a way to specify the vector length.  IIRC we already discussed this elsewhere.  I would expect a TODO i# here for that as we'll need to remove this part of the ifdef.

Reply:
Setting VL for the standalone decoder was supposed to be done as part of https://github.com/DynamoRIO/dynamorio/pull/5776
But I see now that I only enabled it for the disassembler!
https://github.com/DynamoRIO/dynamorio/blob/961580fbec65351380deff46f5448baa88151268/suite/tests/CMakeLists.txt#L1910

Can the same method be used for drdecode?

Reply:
But if you added an interface to set the length isn't that all we need for drdecode?  I see dr_set_sve_vl() (nit: I would have voted for expanding "vl" to "length" or possibly "vector_length"; I guess to me "vl" is not intuitive).

Reply:
> But if you added an interface to set the length isn't that all we need for drdecode? I see dr_set_sve_vl()...

Yes, that should be enough.

Reply:
Done. Added `dr_set_sve_vector_length()` and `dr_get_sve_vector_length()` .